wemake_python_styleguide/visitors/ast/complexity/offset.py

```diff
@@ -55,3 +59,21 @@ def visit_line_expression(self, node: ast.AST) -> None:
         """
         self._check_offset(node)
         self.generic_visit(node)
+
+    def visit_async_statement(self, node: ast.AST):
+        """
+        Checks async function definitions offset.
+
+        This is temporary check for async-based expressions, because offset
+        for them isn't calculated properly. We can calculate right version
+        of offset with subscripting 6.
+
+        Read more: https://bugs.python.org/issue29205
+
+        Raises:
+            TooDeepNestingViolation
+
+        """
+        error = 6 if node.col_offset % 4 != 0 else 0
```

Comment:
Not sure what to do with this magic number as it is temporary solution.

Reply:
I like that you are trying to fix the bug, but I do not like the proposed solution.
Here's why:

1. We have this bug in our `tree` object that comes from `flake8`
2. We solve this bug only for this specific visitor
3. Other visitors that might rely on this value will still face this bug

What I suggest instead?
We can create a mechanism for patching nodes that are required to be patch with either business logic or bugfixes. So we will have this bug fixed for all visitors in a single place.

But where we should do it? I guess this should be executed here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/checker.py#L176-L177

Before any other checks are executed. We can call this technique `Patchers` and also have a pipeline of patchers that is executed before any checkers. 

And we can conditionally call this specific patcher on the affected `python` versions. And ignore it in the future releases.

What does this sound to you?

Reply:
Hm, sounds nice. I'll try to implement this fix soon.